aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -0,0 +1,183 @@
+/// This module supports functionality related to code management.
+module aptos_framework::code {
+    use std::string::String;
+    use std::error;
+    use std::signer;
+    use std::vector;
+
+    // ----------------------------------------------------------------------
+    // Code Publishing
+
+    /// The package registry at the given address.
+    struct PackageRegistry has key {
+        /// Packages installed at this address.
+        packages: vector<PackageMetadata>,
+    }
+
+    /// Metadata for a package.
+    struct PackageMetadata has store, copy, drop {
+        /// Name of this package.
+        name: String,
```

Comment:
Why do we need a name for the package?

Reply:
Because that is the package manifest convention. We basically need to model all data here which is also in the Move.toml, as later we want to be able to generate a Move.toml from this.

Reply:
Okay makes sense